### PR TITLE
ThemeSowcase: Add upload theme button for theme showcase.

### PR DESCRIPTION
**Pull request by budzanowski**
### Info

This PR add upload theme button that will be used for uploading themes :) when it is directly possible or start transfer process to make it possible.

Works only for dev env and will probably stay that way until other elements of the puzzle.

Goes straight to upload page so will not work on sites other than 

### testing
Start calypso in dev mode. Go to singlesite with Jetpack and then upload.

### CODE REVIEW

**Comment by folletto** [comment]
Tested in Chrome:

1. Single site .com, button visible — ✅ 
2. Single site Jetpack, button visible — ✅ 
3. Single site Jetpack, upload drag'n'drop —  ✅ 
4. Single site Jetpack, upload click — ✅ 
5. Upload done, Try & Customize — ✅ 
6. Upload fails on .com sites — ✅  _(It's expected at this stage)_

Note: "Back" from the upload complete screen should go back to the full themes list, now it doesn't. I don't think it's for this PR, but if not let's open a separate issue. :)

**Comment by seear** [comment]
When text appears above the list of wpcom themes, it should indicate that:
screen shot 2016-12-13 at 15 39 43

**Comment by seear** [comment]
Code looks great. We just need the different text depending on which list it appears on. And confirmation from @ockham or @folletto that this will fit ok with the split into two lists?

**Comment by budzanowski** [comment]
@seear This component will be used only for JetpackThemes,
WP.com will have their own with proper txt. It is now located here because it will be its default location in double list scenario. It uses `SectionHeader` for displaying with added `Button` for upload.
For WP.com it will be just `SectionHeade`r so basically no code duplication as this is Calypso native component.

**Comment by budzanowski** [comment]
OK need to rethink what I have just written :( nvm the comment above, sry, temporary brain freeze.

File: client/my-sites/themes/single-site.jsx [hunk 1]
```diff
@@ -40,6 +40,7 @@ const SingleSiteThemeShowcaseWithOptions = ( props ) => {
 					'purchase',
 					'activate',
 					'tryandcustomize',
+					'upload',
```
Comment (seear) [inline client/my-sites/themes/single-site.jsx]: Adding 'upload' to the list here means we get it as an option on the ... button for individual themes, which I don't think we want:
screen shot 2016-12-13 at 17 10 50

Comment (folletto) [inline client/my-sites/themes/single-site.jsx]: Nope. Well spotted.

**Pull request merged**